### DISCUSSION TITLE
Not receiving idToken when using pre-made "Microsoft" identity provider

### DISCUSSION THREAD

**Discussion by danielmichalichyn** [zitadel/zitadel#6200] [Q&A]
I'm opening a discussion for this because i don't know if this is a bug or a "feature" :P 
When i use the "Microsoft" template to configure my identity provider the returned token doesn't have an idToken, but if i configure a "Generic OIDC" i do receive a idToken back. This is a bug or it is supposed to be like this? Or i'm configuring my provider with some wrong parameter?

Thanks in advance.

**Comment by fforootd** [comment]
You do not recieve a id_token from zitadel in your app or does zitadel not recieve the id_token from m$?

**Reply by danielmichalichyn** [reply]
Zitadel does not receive the idToken from Microsoft

**Reply by fforootd** [reply]
Got it, what is your problem with this?

I think we call the userinfo endpoint to fetch data from the user.

**Reply by danielmichalichyn** [reply]
If i've use the Microsoft Provider, the function "getClaim(key)" on my post authentication action doesnt work because we don't have the idToken. Right now i'm using the Generic OIDC and everything works fine. That's why i've opened a discussion, if this is to be this way, there is no problem at all, just for my case i'll have to use the Generic provider :)
Thnks!
